### ISSUE TITLE
[3E11] Update found but "No URL Specified" upon trying to update

### ISSUE THREAD

**Issue by akuropka** [net00-1/SW-DLT#6]
The shortcut indicates that there's an update available (which is, I think, not the case).
There's no update version indicated.
Updating does not work.

![IMG_2343](https://user-images.githubusercontent.com/22570574/133189577-5e7ba798-15ac-4651-8db9-c3492b369093.PNG)
![IMG_2344](https://user-images.githubusercontent.com/22570574/133189593-8066368b-82d5-49dd-8787-35112f5a0345.PNG)
https://user-images.githubusercontent.com/22570574/133189611-6f0ec835-4999-464e-9921-32c123e26cfb.MP4

**Comment by net00-1** [comment]
You can first try to re-download the shortcut from routinehub.co, there are times shortcuts become corrupt and it could be that the updating script was corrupted. It could also have been some temporary error in routinehub that caused this, I tested it multiple times and could not get it to happen on my end.

**Comment by akuropka** [comment]
Basically it's just for your reference and checking... tried right now again and it did not work.
I have attached a screencast.

https://user-images.githubusercontent.com/22570574/133260725-afa2fb66-a141-44c9-9c3b-ba9cc0dc08d0.MP4

**Comment by net00-1** [comment]
I see you're on iOS 15. I haven't been able to update to iOS 15 beta yet to fix whatever issues popped up from it. Thanks for the video, I'll add this to the list of fixes needed for iOS 15 compatibility.

**Comment by Maetran** [comment]
Short note on that issue: it is a simple typo. In maindict vars.updateScript you wrote ‚routinehub.co’ instead of ‚com‘ @net00-1

**Comment by Maetran** [comment]
You mixed up the link

`
data:text/html, <script> var rhGET = new XMLHttpRequest(); rhGET.open("GET","https://routinehub.co/api/v1/shortcuts/7284/versions/latest", false); rhGET.send(); document.write(rhGET.responseText); </script> 
`

The ‚shortcuts‘ part has to come before the number (RID)

**Comment by Riwak** [comment]
Same issue here. Also on iOS 15 ➕

**Comment by net00-1** [comment]
Does anyone have a chance of testing this version of the shortcut? Seems on iOS 15 the data to check for updates includes html tags which messed up how the shortcut checks the json response.

https://www.icloud.com/shortcuts/851b9ac80b824a7db934759db6fd4d17

This also contains the code from the 3E12 branch (multiple fixes)

**Comment by Riwak** [comment]
Hey @net00-1 👋

Thanks for this first update ! Just tested your link 👆
And the « update available message » seem’s solved.

But I tried with this url (https://youtu.be/hgiYw0eG3ps) on my clipboard and got an error message when the shortcut try to launch the py script. 

Cf screen bellow

**Comment by Riwak** [comment]
Hey @net00-1  👋

Thanks for this first update ! Just tested your link 👆
And the « update available message » seem’s solved. 

But I tried with this url (https://youtu.be/hgiYw0eG3ps) in my clipboard and got an error message when the shortcut try to launch the py script. 

Cf screen bellow 
![IMG_1902](https://user-images.githubusercontent.com/24528030/134595991-f4584da7-1f19-4c7b-9785-9846db16386a.PNG)

**Comment by net00-1** [comment]
I see, at least the update issue is solved. I'll keep checking.

**Comment by net00-1** [comment]
I have pinpointed the error to the a-Shell actions in the shortcut and their arguments, can anyone confirm this workaround works?

https://www.icloud.com/shortcuts/625cb8c909db4111a2b937db9a453e88

Basically I replace the list action with text actions which a-shell also supports. This has worked for one tester.

**Comment by Maetran** [comment]
I am having an unspecified error with this version while downloading video and audio. 
sorry can’t be more precise.

**Comment by akuropka** [comment]
@net00-1, works for me, thank you for your efforts!

**Comment by Maetran** [comment]
Complete uninstall and reinstall. Deleteall: True. 
ismini True and false tested. 

Everything’s without effect. Still error.

**Comment by net00-1** [comment]
I see, it definitely is related to the "execute command" action.

I need a way to consistently replicate this bug to find a reliable workaround

**Comment by Maetran** [comment]
Question: isn’t this related to the wrong variable? 
I fand see how the Params are passed to the script. 
![9E63FD09-9754-48A7-83BE-BB08DE1ED0CB](https://user-images.githubusercontent.com/32339445/134780692-12dea2ff-e648-4b38-b2ec-85a00a7c7230.jpeg)

**Comment by net00-1** [comment]
That action you are seeing just grabs the script and saves it inside a-Shell, the "execute command" action that is just below it is the one that sends a command to execute the script with the arguments

https://user-images.githubusercontent.com/53622316/134780864-d7480772-cc79-45bc-82d0-622bf4580d6d.MOV

**Comment by Maetran** [comment]
Ahaaaaa I see. Might be the problem. I have no option to pass the args. It does not accept this option. Strange. 

https://user-images.githubusercontent.com/32339445/134780916-87cc774e-04f5-465c-aec2-2cc43ece6457.MP4

**Comment by Maetran** [comment]
Looks like the style like arts are passed has been changed. 
but passing args there still does not fix the problem. 

![6FBBA738-8378-487B-9D53-CB9372FC1E21](https://user-images.githubusercontent.com/32339445/134781656-9abaed88-416e-4beb-890d-78a8bccb4831.jpeg)

**Comment by akuropka** [comment]
I tested the Shortcut on iOS with a-shell (1.7.6.190, not Mini) on iOS 15.1 and it works out of the box, matters not if parsing the url by sharing or from clipboard. I’m using English (UK) with German region on 12 Pro Max.

**Comment by net00-1** [comment]
Can anyone try this version? I'm testing if it could be a problem with magic variables

https://www.icloud.com/shortcuts/10e57f36803449bbacbf9dc64da86ae6

**Comment by Riwak** [comment]
Still not working on my side. 

https://user-images.githubusercontent.com/24528030/134784548-b7de1d57-b996-4454-848a-5576cb7aa50f.MP4

**Comment by Maetran** [comment]
I tried a bit to fix it and succeeded .  CAREFUL, my link is just for debugging and reconstructing the misbehavior . Don’t download if you don’t want to debug a little ;)

https://www.icloud.com/shortcuts/cbf499ec30ef4b88a2bb11a2235b4a91

Seems like the execution of the shortcut or the a-shell script is too fast. and the providing of the file SW_DLT.py is a little problem. Maybe it is safer or more stable to provide the dictionary and the script on iCloud. Should be an improvement if you ask me.

As well i think you have to re-add the a-shell commands because i think the modified the import.

**Comment by net00-1** [comment]
Yeah, I need to ask the a-Shell dev for any workaround for this. I will update soon.

**Comment by Maetran** [comment]
> Can anyone try this version? I'm testing if it could be a problem with magic variables
> 
> https://www.icloud.com/shortcuts/10e57f36803449bbacbf9dc64da86ae6

I tested your link (from yesterday) now several times and modified just a little - it worked a couple of times now 
It could be useful if you call a script for a-shell at the start of the shortcut.

While testing it felt like the shortcut didn’t ask for permission to communicate with the a-shell extension. calling a random script in a-shell at startup forced the shortcut to ask for permission for communication.
Might be a hint?!

https://www.icloud.com/shortcuts/2503bf84d3c24d7a89885df487ccf2b3

**Comment by net00-1** [comment]
Ok, I finally updated to iOS 15 and spent yesterday/today fixing a bunch of strange bugs. 

The shortcut works reliably for me now, but if you guys are following this issue you can test it before I push the update.

https://www.icloud.com/shortcuts/5bc8bbb10a374941a06aa2ec6237df27

**Comment by akuropka** [comment]
Not working at my end... ):

![IMG_2441](https://user-images.githubusercontent.com/22570574/135029531-7b171f5f-7fa8-4aaf-ab76-2256a590bc33.PNG)

![IMG_2442](https://user-images.githubusercontent.com/22570574/135029561-b607d769-d3d1-4f3f-a5b5-1830d4100d79.PNG)

**Comment by Riwak** [comment]
The last version run smoothly without any issue on my side @net00-1 ! 
Thanks for your work 🙌

**Comment by Maetran** [comment]
same here.
works fine after the _second_ run. at the first run an error occured. but did only 1 test. will come back to you if further bugs occure @net00-1 
thanks for your great work.

**Comment by MarkKropf** [comment]
> Ok, I finally updated to iOS 15 and spent yesterday/today fixing a bunch of strange bugs.
> 
> The shortcut works reliably for me now, but if you guys are following this issue you can test it before I push the update.
> 
> https://www.icloud.com/shortcuts/ef9c278d7b1f4502a9c25bdd71d72755

![C43BA726-4DE4-4950-A5B3-3B5FFE768C53](https://user-images.githubusercontent.com/1703729/135093656-b651323b-33bb-4f67-9623-a0483cf73707.jpeg)

**Comment by MarkKropf** [comment]
Trying my best to capture the error before it skip back to the shortcut app. 
![0C1FA40A-788B-4694-A176-E686FBA168B9](https://user-images.githubusercontent.com/1703729/135094089-07b25ea8-b849-404d-bd56-3394c5be4b25.jpeg)

**Comment by Maetran** [comment]
did you try a clean install? and set flag of "deleteAll" to "true"?
i had to reinstall the dependencies once, after it worked fine.

**Comment by net00-1** [comment]
> Trying my best to capture the error before it skip back to the shortcut app. 
> 
> ![0C1FA40A-788B-4694-A176-E686FBA168B9](https://user-images.githubusercontent.com/1703729/135094089-07b25ea8-b849-404d-bd56-3394c5be4b25.jpeg)
> 
> 

That is an error coming straight from youtube-dl, my script just sends you back to the shortcuts app when an error is raised.

You can try what Maetran said, using the `deleteAll` toggle to delete all dependencies and then installing them again will update them too.

If that doesn't work then the only thing is to wait for the youtube-dl developers to release a patch.

**Comment by MarkKropf** [comment]
Yes, I did set deleteAll to true amongst other things like add/remove ashell, the shortcut, etc. I'll wait to see if youtube-dl gets an update that allows this to work again. Thanks for the quick responses.
